### PR TITLE
Added back-end google api point

### PR DESCRIPTION
**Pull request by brendan-bath**

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -10,20 +10,7 @@
...
-    {
-        "id": 2,
-        "title": "Application Integration"
-    },
-    {
-        "id": 3,
-        "title": "AR and VR"
-    }
-]
+import requests
```
Comment (nat212) [inline backend/app.py:13]: Remove the blank line above.

File: backend/app.py [hunk 2]
```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
             body_token = body.get("token")
             # ensure the id and secret exist
             if body_token:
-                json_val = json.dumps(ec2_instances_mock_data)
+                json_val = json.dumps({"Nothing": "nothing"})
```
Comment (nat212) [inline backend/app.py:143]: Um?

File: backend/app.py [hunk 3]
```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
...
+
+        check_endpoint = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token="
+
+        body = request.get_json()
+
+        if body:
+            auth_email = body.get("email")
+            auth_token = body.get("token")
+            access_key = body.get("access_key")
+            secret_key = body.get("secret_key")
```
Comment (nat212) [inline backend/app.py:168]: I'm thinking we should at some point split the registration into two steps. One where you register your email and password, or use google, then another where you enter your access keys, which are then configurable. But we can worry about that later tbh.

File: backend/app.py [hunk 4]
```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
...
+
+            if all((auth_email, auth_token, access_key, secret_key)):
+                try:
+                    r = requests.get(check_endpoint + auth_token)
```
Comment (nat212) [inline backend/app.py:172]: Rather use `urllib.parse.urlencode`. Look up how to do it.

Comment (nat212) [reply backend/app.py:172]: Then you can pass the auth token in as a data parameter to `requests.get`

File: backend/app.py [hunk 5]
```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
...
+                    data = r.json()
+
+                    if data.get("issued_to") and data.get("issued_to") == auth_email:
+                        token = str(uuid.uuid4())
+                        ttl = 128000
```
Comment (nat212) [inline backend/app.py:177]: Maybe make the ttl a standard global variable, or put it in config or something

File: backend/app.py [hunk 6]
```diff
@@ -153,10 +140,64 @@ def ec2_instances() -> Response:
...
+                            'expires': datetime.now() + timedelta(seconds=ttl),
+                        })
+                        res = {'token': token, 'ttl': ttl}
+                        return Response(json.dumps(res), status=200, mimetype='application/json')
+                except Exception as e:
+                    res = {'error': e}
+                    return Response(json.dumps(res), status=400, mimetype='application/json')
+            else:
+                return Response(json.dumps({
+                    'message': f"Missing fields: email, token, access_key, or secret_key"
```
Comment (nat212) [inline backend/app.py:198]: Why did you make it an f-string if there are no variables?

File: backend/tests/setup.cfg [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+max-line-length = 120
```
Comment (nat212) [inline backend/tests/setup.cfg:2]: Why did you make this file?

File: backend/tests/test_google_verify.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
+from unittest import TestCase
```
Comment (nat212) [inline backend/tests/test_google_verify.py:1]: In my PR, I made a standardized factory for creating endpoint test classes. Once that's merged in, we need to convert this one as well.

**Review by nat212 — CHANGES_REQUESTED** [review]

File: backend/app.py (hunk 2 above)
Comment (brendan-bath) [reply backend/app.py:143]: Its for the old ec2 instance list just a dummy value because it is being removed

File: backend/app.py (hunk 5 above)
Comment (brendan-bath) [reply backend/app.py:177]: I'll make it a ttl a global for now

File: backend/tests/setup.cfg (hunk 7 above)
Comment (brendan-bath) [reply backend/tests/setup.cfg:2]: I was getting an error when running tests that a cfg file was missing.

File: backend/tests/test_google_verify.py (hunk 8 above)
Comment (brendan-bath) [reply backend/tests/test_google_verify.py:1]: sure

File: backend/app.py (hunk 6 above)
Comment (brendan-bath) [reply backend/app.py:198]: not sure will remove

**Pull request merged**